src/burn/engine/plan.cpp

```diff
@@ -625,7 +625,11 @@ extern "C" HRESULT PlanRegistration(
     UINT cDependencies = 0;
 
     pPlan->fRegister = TRUE; // register the bundle since we're modifying machine state.
-    pPlan->fKeepRegistrationDefault = (pRegistration->fInstalled || BOOTSTRAPPER_RESUME_TYPE_REBOOT == resumeType); // keep the registration if the bundle was already installed or we are planning after a restart.
+
+    // Keep the registration if the bundle was already installed or we are planning after a restart.
+    // Also keep the registration if we are caching because the ARP (via uninstall) allows uncaching.
```

Comment:
"Also keep the registration..." should be removed, right?  This comment isn't in wix3.